torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -57,6 +57,57 @@ std::string ncclGetErrorWithVersion(ncclResult_t error) {
       getNcclVersion();
 }
 
+// Provides additional detail into NCCL error codes based on when these are
+// thrown in the NCCL codebase.
+std::string getNcclErrorDetailStr(
+    ncclResult_t error,
+    c10::optional<std::string> processGroupFailureReason /* = c10::nullopt */
+) {
+  // Prioritize failure reason provided by PG NCCL first, as it can abort
+  // communicators when it encounters collective timeouts, etc.
+  if (processGroupFailureReason != c10::nullopt) {
+    return (*processGroupFailureReason).c_str();
```

Comment:
nit: we aren't return c string now, so we can just return `(*processGroupFailureReason)`

Reply:
Removed. Thanks!